exporters/elasticsearch-exporter/src/main/java/io/zeebe/exporter/ElasticsearchClient.java

```diff
@@ -136,21 +140,36 @@ public boolean flush() {
         bulkRequest = new ArrayList<>();
       }
     }
-
     return success;
   }
 
-  private boolean exportBulk() throws IOException {
+  private boolean checkBulkResponse(final BulkResponse bulkResponse) {
+    final var hasErrors = bulkResponse.hasErrors();
+    if (hasErrors) {
+      bulkResponse.getItems().stream()
+          .map(BulkItem::getIndex)
+          .map(BulkItemIndex::getError)
+          .forEach(
+              error ->
+                  log.warn(
+                      "Failed to flush item of bulk request [type: {}, reason: {}]",
```

Comment:
@saig0 I don't think it is a good idea to log every failed item, the reason is that we have thousands of records in a bulk request in a high load scenario. As far as I remember initial we had a similar approach in the first implementation which we saw as a potential problem with log pollution, see https://github.com/zeebe-io/zeebe/pull/1344#discussion_r218088251.

/cc @npepinpe wdyt?

Reply:
This is again a case where we would benefit from having a burst filter for errors, a way to log an error max X times over Y seconds; we have several other places where we could benefit from this.

As it is I agree, if ES is down this will create tons of logs, and it just creates noise. Suppressing it however hides an issue, so...maybe we can find a middle ground? Log the error every X errors?

Reply:
As a side note we also had the topic yesterday in the cloud product meeting that Operate is logging to much in case elastic is unavailable, this is also which would trigger an immense log burst with this code.

I would highly recommend to not release this as it is right now as this will immediately lead to problems in cloud.

Reply:
Agree here with @menski maybe we can have a metric which count flash fails or something similar. And yes something like a burst filter would be nice.

Reply:
Agrona has a counters buffer which keeps track of errors and how often they happen.I guess this works well if each error gets a specific code or something, not sure how they deal with context. That said, I also don't think it plays well with logging systems; my point here is that it's definitely a common problem, so I'm sure there's a solution [out there](https://i.imgur.com/nHghxAr.gif)

Reply:
If ES is not reachable then the log statement is not printed. Instead, an exception is thrown and the exporter gets back-off.

To avoid too much log messages, I could do the following:
* log message only once per type for one bulk flush operation
* throw an exception if the flush has errors to get back-off

What do you think about these changes?

Reply:
Let's go for that :+1:

Reply:
Will be fixed by #4732